lib/base-service.ts

```diff
@@ -255,6 +277,31 @@ export class BaseService {
     );
   }
 
+  /**
+   * Wrapper around `createRequest` that enforces arrived response to be deserialized.
+   * @param parameters - see `parameters` in `createRequest`
+   * @param deserializerFn - the deserializer function that is applied on the response object
+   * @param isMap - is `true` when the response object should be handled as a map
+   * @protected
+   * @returns a Promise
+   */
+  protected createRequestAndDeserializeResponse(
```

Comment:
The name of this function is `createRequestAndDeserializeResponse`, but yet it ends up calling a function named `serializeModel` after receiving the response body.  Shouldn't this be called `deserializeModel` instead?   Serialization would be like calling JSON.stringify() whereas deserialization would be like calling JSON.parse().

Reply:
I agree, I renamed the function to the less confusing `convertModel` in [9c9e5cf](https://github.com/IBM/node-sdk-core/pull/247/commits/9c9e5cf4dfcf13dea654a284c62587042806f7ec).